### PR TITLE
Add argument & command validation in scope of repo

### LINKED ISSUE #233: Unknown preset is silently defaulted to a root command and sends wrong default attributes
Unknown preset is silently defaulted to a root command and sends wrong default attributes.
Not sure how `bot try-runtime trappist` ended up with [this](https://github.com/paritytech/trappist/pull/259#issuecomment-1685964956) but it sends the default arguments mixed with trappist one
`--chain=polkadot --chain_node=polkadot --target_path=. --live_uri=rococo-trappist`
while in docs it shows 

> Default static args:
> chain: trappist
chain_node: trappist-node
target_path: .
live_uri: rococo-trappist

In scope of this i identified that you can also 
- run command which is not expected to be run in this repo
- run command with unexisting preset like `bot try-runtime ololo` and it will still run it

```[tasklist]
### Tasks
- [x] generate commands only in scope of repo
- [x] validate the preset name, otherwise return error
```

### PR DESCRIPTION
**Pull request by mordamax**
Closes #233 
Tested locally 
image

Also added validation of commands & subcommands in scope of repo - will fail with related message

### CODE REVIEW

File: src/command-configs/utils.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      includesGenericPresets = true;
+    }
+
+    for (const preset of presets) {
+      const repos = preset.repos ?? [];
+      // if repos are not specified, then this command is supported by all repos
+      if (repos.length === 0) {
+        includesGenericPresets = true;
+      }
+      for (const repo of preset.repos ?? []) {
```
Comment (mutantcornholio) [inline src/command-configs/utils.ts:24]:
```suggestion
      for (const repo of repos) {
```

File: src/commander/commander.ts [hunk 2]
```diff
@@ -38,6 +39,38 @@ export function getCommanderFromConfiguration(
   let parsedCommand: ParseResults["parsedCommand"] = undefined;
   const helpStr = `Refer to [help docs](${docsPath}) and/or [source code](${config.pipelineScripts.repository}).`;
 
+  const exitOverride = (commandKey: string) => (e: CommanderError) => {
+    if (e.code === "commander.excessArguments") {
+      throw new Error(((e as CommanderError).message = `Unknown subcommand of "${commandKey}". ${helpStr}`));
```
[the remark this answers is not in the document]
Comment (mutantcornholio) [reply src/commander/commander.ts:44]: Yeah, by the looks of it 
```ts
throw new Error(((e as CommanderError).message = `Unknown subcommand of "${commandKey}". ${helpStr}`));
```
is equivalent to 
```ts
throw new Error(`Unknown subcommand of "${commandKey}". ${helpStr}`);
```

File: src/bot/parse/parsePullRequestBotCommandLine.namedArgs.spec.ts [hunk 3]
```diff
@@ -184,14 +210,14 @@ const dataProvider: DataProvider[] = [
   { suitName: "clear", commandLine: "bot clear", expectedResponse: new CleanCommand() },
 
   /*
-    Cancel cases
-   */
+      Cancel cases
+     */
```
[the remark this answers is not in the document]
Comment (mordamax) [reply src/bot/parse/parsePullRequestBotCommandLine.namedArgs.spec.ts:214]: most probably :)

File: src/commander/commander.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mordamax) [reply src/commander/commander.ts:44]: haha i just copied from the bottom one, and refactored in really ugly way not even thinking 😂😂

thanks, fixed!

**Pull request merged**